assets/aml-benchmark/components/src/utils/aml_run_utils.py

```diff
@@ -0,0 +1,97 @@
+
+"""Helper functions for AML runs."""
+from typing import List, Optional, Tuple, cast
+import os
+import tempfile
+import re
+
+import mlflow
+from mlflow.entities import Run as MLFlowRun
+from azureml.core import Run
+from utils.logging import get_logger
+
+
+logger = get_logger(__name__)
+
+
+def get_experiment_name() -> str:
+    """Get the current experiment name."""
+    return Run.get_context().experiment.name
+
+
+def get_parent_run_id() -> str:
+    """Get the run id of the parent of the current run."""
+    return cast(str, Run.get_context().parent.id)
+
+
+def get_all_runs_in_current_experiment() -> List[MLFlowRun]:
+    """
+    Get a list of all of the runs in the current experiment \
+    that are a direct child of the root run except the current run.
+
+    :returns: The list of runs in the current experiment.
+    """
+    experiment_name = get_experiment_name()
+    parent_run_id = get_parent_run_id()
+    runs = cast(List[MLFlowRun], mlflow.search_runs(
+        experiment_names=[experiment_name],
+        filter_string=f"tags.mlflow.parentRunId='{parent_run_id}'",
+        output_format='list'
+    ))
+    return [
+        run for run in runs
+        if run.info.run_id != Run.get_context().id and run.info.run_id != parent_run_id
+    ]
+
+
+def get_compute_information(log_files: Optional[List[str]], run_v1: Run) -> Optional[str]:
+    """Get the VMType used for a given run."""
+    if log_files is not None:
+        file_name = 'execution-wrapper.log'
+        complete_file_name = 'system_logs/lifecycler/' + file_name
+        if complete_file_name in log_files:
+            try:
+                with tempfile.TemporaryDirectory() as tmp:
+                    run_v1.download_file(complete_file_name, tmp)
+                    with open(os.path.join(tmp, file_name), 'r') as file:
+                        content = file.read()
+                        compute_match = re.search(
+                            r'vm_size: Some\("([^\"]*)\"',
+                            content,
+                        )
+                if compute_match:
+                    return compute_match.group(1)
+            except Exception as ex:
+                logger.warn(f"Failed to get system_logs/lifecycler/execution-wrapper.log \
+                             due to {ex}")
+                return None
+
+
+def get_step_name(run: MLFlowRun) -> str:
+    """Get the step name of a given run."""
+    stepName = 'stepName'
+    if stepName in run.data.tags:
+        return run.data.tags[stepName]
+    return run.info.run_name
+
+
+def get_mlflow_model_name_version(model_uri: str) -> Tuple[str, Optional[str], Optional[str]]:
+    """
+    Get the mlflow model name and version from a URI.
+
+    :param model_uri: the URI from which the model name and version has to be parsed.
+
+    :returns: The model name and version.
+    """
+    # If the model is from a registry, parse the information from URI
+    if model_uri.startswith("azureml://registries/"):
```

Comment:
could this write as `azureml:resgistries/...`?

Reply:
I tried passing `azureml:registries/azureml/models/finiteautomata-bertweet-base-sentiment-analysis/versions/8`. It threw an error complaining- `If providing an ARM id, it should start with a '/'.`. Hence, `//` is required for registry asset (Single `/` also fails).